crates/aptos/src/common/utils.rs

```diff
@@ -340,19 +344,22 @@ pub fn generate_vanity_account_ed25519(
     hex::decode(to_check_if_is_hex).  // Check that the vanity prefix can be decoded into hex.
         map_err(|error| CliError::CommandArgumentError(format!(
             "The vanity prefix could not be decoded to hex: {}", error)))?;
-    // Create a random key generator based on OS random number generator.
-    let mut key_generator = KeyGen::from_os_rng();
-    // Randomly generate a new Ed25519 private key.
-    let mut private_key = key_generator.generate_ed25519_private_key();
-    // While the account address does not start with the vanity prefix:
-    while !account_address_from_public_key(&Ed25519PublicKey::from(&private_key))
-        .short_str_lossless()
-        .starts_with(vanity_prefix_ref)
-    {
-        // Generate a new account.
-        private_key = key_generator.generate_ed25519_private_key();
+    let mut key_generator = KeyGen::from_os_rng(); // Get random key generator.
+    loop {
+        // Generate new keys until finding a match against the vanity prefix.
+        let private_key = key_generator.generate_ed25519_private_key();
```

Comment:
Is this susceptible at all to people getting the same key when asking for the same prefix? This of course highly depend on the seed used in the generation process but I wonder if this search adds bias here that can increase the probability

Reply:
The random key generator used here is the same as that used for standard key generation.

Hence the probability of a "collision" is the same as for using the CLI for key generation without a prefix.